src/java/org/apache/cassandra/index/sai/disk/format/IndexDescriptor.java

```diff
@@ -384,34 +384,30 @@ public boolean validatePerIndexComponents(IndexTermType indexTermType, IndexIden
     }
 
     @SuppressWarnings("BooleanMethodIsAlwaysInverted")
-    public boolean validatePerSSTableComponents(IndexValidation validation)
+    public boolean validatePerSSTableComponents(IndexValidation validation, boolean rethrow)
     {
         if (validation == IndexValidation.NONE)
             return true;
 
-        logger.info(logMessage("Validating per-sstable index components using mode " + validation));
-        boolean checksum = validation == IndexValidation.CHECKSUM;
+        logger.info(logMessage("Validating per-sstable index components for SSTable {} using mode {}"), sstableDescriptor.toString(), validation);
 
         try
         {
-            version.onDiskFormat().validatePerSSTableIndexComponents(this, checksum);
+            version.onDiskFormat().validatePerSSTableIndexComponents(this, validation == IndexValidation.CHECKSUM);
             return true;
         }
         catch (UncheckedIOException e)
         {
-            return false;
+            if (rethrow)
+                throw e;
+            else
+                return false;
         }
     }
 
-    public void checksumPerIndexComponents(IndexTermType indexTermType, IndexIdentifier indexIdentifier)
-    {
-        version.onDiskFormat().validatePerColumnIndexComponents(this, indexTermType, indexIdentifier, true);
-    }
-
-    @SuppressWarnings("BooleanMethodIsAlwaysInverted")
-    public void checksumPerSSTableComponents()
+    public void checksumPerIndexComponents(IndexTermType indexTermType, IndexIdentifier indexIdentifier, boolean validateChecksum)
```

Comment:
We could just remove this method and modify `validatePerIndexComponents()` to take a `rethrow` argument (similar to `validatePerSSTableComponents()`)?